entrypoint.sh

```diff
@@ -67,30 +83,46 @@ echo "========================================================================"
 
 for ((i=0;i<20;i++))
 do
-    DB_CONNECTABLE=$(mysql -u"$DB_USER" -p"$DB_PASS" -h"$DB_HOST" -P"$DB_PORT" -e 'status' >/dev/null 2>&1; echo "$?")
-    if [[ $DB_CONNECTABLE -eq 0 ]]; then
-        break
+    if [[ $DB_DRIVER == "mysql" ]]; then
+        DB_CONNECTABLE=$(mysql -u"$DB_USER" -p"$DB_PASS" -h"$DB_HOST" -P"$DB_PORT" -e 'status' >/dev/null 2>&1; echo "$?")
+        if [[ $DB_CONNECTABLE -eq 0 ]]; then
+            break
+        fi
+    else
+        DB_CONNECTABLE=$(psql -U "$DB_USER" -h "$DB_HOST" -p "$DB_PORT" -l >/dev/null 2>&1; echo "$?")
+        if [[ $DB_CONNECTABLE -eq 0 ]]; then
+            break
+        fi
     fi
     sleep 3
 done
 
 if ! [[ $DB_CONNECTABLE -eq 0 ]]; then
-	echo "Cannot connect to MySQL"
+	echo "Cannot connect to database"
     exit $DB_CONNECTABLE
 fi
-
 ###
 
-if ! drush sql-query "SHOW DATABASES LIKE '${DB_NAME}';" > /dev/null ; then
-	run_scripts setup
-    echo "=> Done installing site!"
-	if [ $EXTRA_SETUP_SCRIPT ]; then
-		echo "=> WARNING: The usage of EXTRA_SETUP_SCRIPT is deprectated. Put your script into /scripts/post-setup.d/"
-		. $EXTRA_SETUP_SCRIPT
-		echo "=> Successfully ran extra setup script ${EXTRA_SETUP_SCRIPT}."
+if [[ $DB_DRIVER == "mysql" ]]; then
+	if ! drush sql-query "SHOW DATABASES LIKE '${DB_NAME}';" > /dev/null ; then
+		run_scripts setup
+	    echo "=> Done installing site!"
+		if [ $EXTRA_SETUP_SCRIPT ]; then
+			echo "=> WARNING: The usage of EXTRA_SETUP_SCRIPT is deprectated. Put your script into /scripts/post-setup.d/"
+			. $EXTRA_SETUP_SCRIPT
+			echo "=> Successfully ran extra setup script ${EXTRA_SETUP_SCRIPT}."-]
+		fi
+	else
+	    echo "=> Skipped setup - database ${DB_NAME} already exists."
+	fi
+elif [[ $DB_DRIVER == "pgsql" ]]; then
+	drush sql-query --result-file='table-query.txt' '\dt';
```

Comment:
no, it returns an empty query when there are no results, hence the `wc` below. yes, this is true, it just looks for any table. That's the correct behaviour, as you'll very likely already be inside of a database, using the `public` (i.e. default) schema. You're just checking for tables, because that's approximately as good of a check as checking for the presence of a database in mysql. I'll change this to look for a drupal specific table shortly.